### ISSUE TITLE
sendmail-auth.conf _daemon does not match on Fedora/RHEL

### ISSUE THREAD

**Issue by opoplawski** [fail2ban/fail2ban#1632]
### Environment:

- OS, including release name/version: Fedora 24
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
### The issue:

filter.d/sendmail-auth.conf has:
```
_daemon = (?:sm-(mta|acceptingconnections))
```
This does not match on Fedora/RHEL - the daemon name there is "sendmail"

See also https://bugzilla.redhat.com/show_bug.cgi?id=1329919

I'm not really sure what the proper regex should be here to preserve the same matching.

**Comment by sebres** [comment]
Either the Fedora/RHEL distribution should get another config (or should overwrite it somewhere in after "includes", like `jail.d/...`) to `_daemon = sendmail`
or build a compatible expression (I prefer this, cause it is essentially fast enough):
```diff
- _daemon = (?:sm-(mta|acceptingconnections))
+ _daemon = (?:sendmail|sm-(?:mta|acceptingconnections))
```

**Comment by K2DLS** [comment]
For FC25 with journalmatch, the _daemon line is not necessary.

**Comment by sebres** [comment]
Final `__daemon_re` what used in the `__prefix_line` expression is optional.
See https://github.com/fail2ban/fail2ban/blob/0.10/config/filter.d/common.conf#L25

But in principle it can be set to empty value, if systemd backend used.

**Comment by K2DLS** [comment]
Working now on FC25 with package 0.9.6-2, also pushed to FC24 and RHEL.  This can be closed.

fail2ban-regex "systemd-journal" /etc/fail2ban/filter.d/sendmail-auth.conf

Running tests
=============

Use   failregex filter file : sendmail-auth, basedir: /etc/fail2ban
Use         systemd journal
Use         encoding : UTF-8
Use    journal match : _SYSTEMD_UNIT=sendmail.service

Results
=======

Failregex: 66 total
|-  #) [# of hits] regular expression
|   1) [66] ^(?:\[\])?\s*(?:<[^.]+\.[^.]+>\s+)?(?:\S+\s+)?(?:kernel: \[ *\d+\.\d+\]\s+)?(?:@vserver_\S+\s+)?(?:(?:(?:\[\d+\])?:\s+[\[\(]?(?:sendmail|sm-(?:mta|acceptingconnections))(?:\(\S+\))?[\]\)]?:?|[\[\(]?(?:sendmail|sm-(?:mta|acceptingconnections))(?:\(\S+\))?[\]\)]?:?(?:\[\d+\])?:?)\s+)?(?:\[ID \d+ \S+\]\s+)?\w{14}: (\S+ )?\[<HOST>\]( \(may be forged\))?: possible SMTP attack: command=AUTH, count=\d+$
`-
Ignoreregex: 0 total

**Comment by sebres** [comment]
fixed in 314e402fe0fd46b102d94000ec626b212bdca657
